arch/arc/core/thread.c

```diff
@@ -134,14 +134,16 @@ void _new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 		 _ARC_V2_STATUS32_E(_ARC_V2_DEF_IRQ_LEVEL);
 #ifdef CONFIG_USERSPACE
 	if (options & K_USER) {
-		thread->arch.u_stack_top = (u32_t)pStackMem;
+		/* +4 bytes to reserve place for errno */
+		thread->arch.u_stack_top = (u32_t)pStackMem + 4;
 		thread->arch.u_stack_base = (u32_t)stackEnd;
 		thread->arch.k_stack_top =
 			 (u32_t)(stackEnd + STACK_GUARD_SIZE);
 		thread->arch.k_stack_base = (u32_t)
 		(stackEnd + STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE);
 	} else {
-		thread->arch.k_stack_top = (u32_t)pStackMem;
+		/* +4 bytes to reserve place for errno */
+		thread->arch.k_stack_top = (u32_t)pStackMem + 4;
```

Comment:
Should errno_location be updated here since it's in the guard space (if MPU stack guards would be used)?